kafka-client/src/main/java/dev/responsive/db/CassandraIdempotentKeyValueSchema.java

```diff
@@ -0,0 +1,289 @@
+   * @param partitionKey  the partitioning key
+   * @param key           the data key
+   *
+   * @return a statement that, when executed, will delete the row
+   *         matching {@code partitionKey} and {@code key} in the
+   *         {@code table}
+   */
+  @Override
+  @CheckReturnValue
+  public BoundStatement delete(
+      final String table,
+      final int partitionKey,
+      final Bytes key
+  ) {
+    return delete.get(table)
+        .bind()
+        .setByteBuffer(DATA_KEY.bind(), ByteBuffer.wrap(key.get()));
+  }
+
+
+  /**
+   * Inserts data into {@code table}. Note that this will overwrite
+   * any existing entry in the table with the same key.
+   *
+   * @param table         the table to insert into
+   * @param partitionKey  the partitioning key
+   * @param key           the data key
+   * @param value         the data value
+   *
+   * @return a statement that, when executed, will insert the row
+   *         matching {@code partitionKey} and {@code key} in the
+   *         {@code table} with value {@code value}
+   */
+  @Override
+  @CheckReturnValue
+  public BoundStatement insert(
+      final String table,
+      final int partitionKey,
+      final Bytes key,
+      final byte[] value
+  ) {
+    return insert.get(table)
+        .bind()
+        .setByteBuffer(DATA_KEY.bind(), ByteBuffer.wrap(key.get()))
+        .setByteBuffer(DATA_VALUE.bind(), ByteBuffer.wrap(value));
+  }
+
+  /**
+   * Retrieves the value of the given {@code partitionKey} and {@code key}
+   * from {@code table}.
+   *
+   * @param tableName the table to retrieve from
+   * @param partition the partition
+   * @param key       the data key
+   *
+   * @return the value previously set
+   */
+  @Override
+  public byte[] get(final String tableName, final int partition, final Bytes key) {
+    final BoundStatement get = this.get.get(tableName)
+        .bind()
+        .setByteBuffer(DATA_KEY.bind(), ByteBuffer.wrap(key.get()));
+
+    final List<Row> result = client.execute(get).all();
+    if (result.size() > 1) {
+      throw new IllegalArgumentException();
+    } else if (result.isEmpty()) {
+      return null;
+    } else {
+      final ByteBuffer value = result.get(0).getByteBuffer(DATA_VALUE.column());
+      return Objects.requireNonNull(value).array();
+    }
+  }
+
+  @Override
+  public KeyValueIterator<Bytes, byte[]> range(
+      final String tableName,
+      final int partition,
+      final Bytes from,
+      final Bytes to
+  ) {
+    throw new UnsupportedOperationException("range scans are not supported on Idempotent schemas.");
+  }
+
+  @Override
+  public KeyValueIterator<Bytes, byte[]> all(
+      final String tableName,
+      final int partition
+  ) {
+    throw new UnsupportedOperationException("all is not supported on Idempotent schemas");
+  }
+
+  private static ByteBuffer metadataKey(final int partition) {
+    final ByteBuffer key = ByteBuffer.wrap(new byte[METADATA_KEY.get().length + Integer.BYTES]);
```

Comment:
I think it needs to be the same type as the key, which is blob - I found this relatively easy to debug with a small tool to generate the hex and then select from C* using `SELECT * FROM table WHERE dataKey = 0x<theHex>`.

I could play around with the format so that we could use `TextAsBlob` in C* for selecting, but I don't think we can make it a string.